internal/mode/static/telemetry/collector.go

```diff
@@ -119,12 +133,17 @@ func NewDataCollectorImpl(
 
 // Collect collects and returns telemetry Data.
 func (c DataCollectorImpl) Collect(ctx context.Context) (Data, error) {
+	g := c.cfg.GraphGetter.GetLatestGraph()
+	if g == nil {
+		return Data{}, errors.New("failed to collect telemetry data: latest graph cannot be nil")
+	}
+
```

Comment:
why do we need to check for nil? 

or 

good catch if it was an edge case you caught!

Reply:
this is just moving it out of the `collectGraphResourceCount` function and extracting it up here so we can just pass the graph into the new `collectSnippetsFilterDirectives` and older `collectGraphResourceCount`. So it already existed, and I think it was added because `Collect` could be called before NGF processes things and produces an actual graph, and if the graph isn't populated then there is no reason to collect data.